lib/CGameState.cpp

```diff
@@ -3541,3 +3554,48 @@ CRandomGenerator & CGameState::getRandomGenerator()
 	//logGlobal->traceStream() << "Fetching CGameState::rand with seed " << rand.nextInt();
 	return rand;
 }
+
+bool CPathfinder::addTeleportTwoWay(const CGTeleport * obj) const
+{
+	if(allowTeleportTwoWay)
+	{
+		if(ETeleportChannelType::BIDIRECTIONAL == gs->getTeleportChannelType(obj->channel, hero->tempOwner))
+			return true;
+	}
+	return false;
+}
```

Comment:
> Is this one like really needed?

I don`t insist on changing this particular place. OTOH you may add isTeleportBiDirectional() to gs and make line shorter.